src/NuGet.Core/NuGet.Commands/CommandRunners/PackCommandRunner.cs

```diff
@@ -642,7 +642,11 @@ private PackageArchiveReader BuildFromNuspec(string path)
 
             if (packageArchiveReader != null && !_packArgs.NoPackageAnalysis)
             {
-                AnalyzePackage(packageArchiveReader);
+                var iconRuletset = new IPackageRule[] {
```

Comment:
Because:
-  if I add it to the collection, I have no way to change the warning message dynamically
- There's no way to know where do you are generating the `PackageArchiveReader` from `IPackageRule.Validate()`

Reply:
Hmm...I get your scenario. 
The trick is the Rules set is something others can depend on for validations. 

I don't necessarily have an idea off the top of my head, but I would hate for us to create unnnecessary objects.

Reply:
I will add a generic message:

The 'iconUrl'/'PackageIconUrl' element is deprecated. Consider using the 'icon'/'PackageIcon' element instead.